src/dali_executor/dali_executor.h

```diff
@@ -42,9 +42,12 @@ struct shape_and_type_t {
 class DaliExecutor {
  public:
   DaliExecutor(
-      std::string serialized_pipeline, int max_batch_size, int device_id)
+      std::string serialized_pipeline, int max_batch_size, int num_threads, int device_id)
       : serialized_pipeline_(std::move(serialized_pipeline)),
-        device_id_(device_id), max_batch_size_(max_batch_size)
+        max_batch_size_(max_batch_size),
```

Comment:
If currently there's 1-1 relation `DaliExecutor` to `DaliPipeline`, do we need `serialized_pipeline_` here? I'd consider designing it so that `DaliExecutor` just gets any `DaliPipeline`. After all, the executor doesn't have to know at all what's inside `DaliPipeline`, right? Just needs to run something through it

Reply:
I moved the creation of pipeline out of executor and moved Resetting logic to pipeline.